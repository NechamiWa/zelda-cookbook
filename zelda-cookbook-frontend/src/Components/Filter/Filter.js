import React, { Component } from 'react';
import IngredientFilter from '../../Containers/IngredientFilter/IngredientFilter';
import NameFilter from '../../Containers/NameFilter/NameFilter';
import EffectFilter from '../../Containers/EffectFilter/EffectFilter';
import './Filter.css';

export class Filter extends Component {
  render() {
    return (
      <div className='Filter'>
        <h2 className='filter__category' >
          Recipe Name:
        </h2>
        <NameFilter />
        <h2 className='filter__category'>
          Type:
        </h2>
        <EffectFilter />
        <h2 className='filter__category'>
          Ingredients:
        </h2>
        <IngredientFilter />
      </div>
    );
  }
}

export default Filter;
