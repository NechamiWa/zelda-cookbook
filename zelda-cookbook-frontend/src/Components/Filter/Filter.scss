.Filter {
  margin: 30px 0 0;
}

.search-bar {
  font-size: 1.4em;
  width: 240px;
  border: 1px solid black;
  padding: 5px;
  border-radius: 0;
}

.search-bar__ingredient {
  display: inline-block;
  position: relative;
}

.filter__category {
  font-family: fantasy, 'sans-serif';
  letter-spacing: 1px;    
  text-align: left;
  font-size: 1.5em;
  color: #EDE5CB;
}

.submit-button {
  font-size: 1.4em;
  border: 1px solid black;
}

.search-results__container {
  width: 300px;
  border: 1px solid black;
  display: none;
  position: absolute;
  max-height: 260px;
  overflow: scroll;
}

.search-results__containter--show {
  display: block;
}

.search-results__item {
  margin: 0;
  padding: 15px;
  font-size: 1.4em;
  background-color: rgba(255, 255, 255, .9);

  &:hover {
    background-color: gray;
    font-weight: bold;
  }
}

.NameFilter,
.IngredientTag,
.EffectFilter {
  display: grid;
  grid-template-columns: 1fr 40px;
  align-items: center;
}

.button__clear-filter {
  height: 40px;
  width: 40px;
  display: inline-block;
  font-size: 20px;
  color: #1c3744;
  text-align: center;
}

.button__clear-filter--hidden {
  display: none;
}