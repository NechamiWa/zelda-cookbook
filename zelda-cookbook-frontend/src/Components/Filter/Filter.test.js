/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import Filter from './Filter';

describe('Filter', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<Filter />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });  
});