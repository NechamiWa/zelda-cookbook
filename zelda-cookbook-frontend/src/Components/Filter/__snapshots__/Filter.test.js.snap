// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should match the snapshot 1`] = `
<div
  className="Filter"
>
  <h2
    className="filter__category"
  >
    Recipe Name:
  </h2>
  <Connect(NameFilter) />
  <h2
    className="filter__category"
  >
    Type:
  </h2>
  <Connect(EffectFilter) />
  <h2
    className="filter__category"
  >
    Ingredients:
  </h2>
  <Connect(IngredientFilter) />
</div>
`;
